src/datadog/grok/filters/keyvalue.rs

```diff
@@ -156,72 +173,50 @@ fn regex_from_config(
     Regex::new(keyvalue.as_str()).ok()
 }
 
-#[derive(Debug, Clone)]
-pub struct KeyValueFilter {
-    pub re_pattern: Regex,
-    pub quotes: Vec<(char, char)>,
-    pub key_value_delimiter: String,
-}
-
-impl std::fmt::Display for KeyValueFilter {
-    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        write!(
-            f,
-            "keyvalue(\"{:?}\", \"{:?}\")",
-            self.re_pattern, self.quotes
-        )
-    }
-}
-
-pub fn apply_filter(value: &Value, filter: &KeyValueFilter) -> Result<Value, GrokRuntimeError> {
-    match value {
-        Value::Bytes(bytes) => {
-            let mut result = Value::Object(BTreeMap::default());
-            let value = String::from_utf8_lossy(bytes);
-            filter
-                .re_pattern
-                .captures_iter(value.as_ref())
-                .for_each(|c| {
-                    parse_key_value_capture(filter, &mut result, c);
+impl KeyValueFilter {
```

Comment:
is the motivation behind having two seperate implementations organizational? i.e. first one concerning itself with initialization and this one with processing

Reply:
Pretty much just minimizing the code movement in the diff to make it (a little) easier to review. I can move it if you think it's worth it.

Reply:
Ah gotcha 👍 feels fine as is, I just hadn't seen that much yet so I was mainly curious